### PR TITLE
[next-css] Provides Manifest Hash in Production

### PR DESCRIPTION
**Pull request by bhongy**
In production, extract style.css with content hash to leverage long-term browser caching.

As https://github.com/zeit/next.js/pull/4119 landed, now we have access to webpack manifest in `_document.js` via `this.props.buildManifest`. This PR enables build with contenthash in `!dev`.

Related to issue: https://github.com/zeit/next-plugins/issues/11

Test was successful on my personal site (inspect the network tab): https://thanik.me

### CODE REVIEW

**Comment by bhongy** [comment]
I only updated next-css (I don't use the sass, stylus or postcss ones). Pushing to get some feedback but I won't mind updating those as well.

File: packages/next-css/readme.md [hunk 1]
```diff
@@ -195,6 +195,35 @@ Create a CSS file `style.css` the CSS here is using the css-variables postcss pl
...
+import Document, { Head, Main, NextScript } from 'next/document'
+
+export default class MyDocument extends Document {
+  render() {
+    const { buildManifest } = this.props;
+    return (
+      <html>
+        <Head>
+          {buildManifest.css.map(assetPath => (
+            // this will work in both development and production
```
Comment (timneutkens) [inline packages/next-css/readme.md]: Sounds like this comment is contradicting the earlier mention you made:

> For production builds, the hashed filename is available in `this.props.buildManifest.css`, so the `_document.js` could be changed to:

To be clear buildManifest works in both development and production mode as far as I know 👍

Comment (bhongy) [reply packages/next-css/readme.md]: Agreed, it is a bit confusing. I rewrote it 079a4fa. Let me know if it's still confusing. :)

File: packages/next-css/readme.md [hunk 2]
```diff
@@ -195,6 +195,34 @@ Create a CSS file `style.css` the CSS here is using the css-variables postcss pl
 
 When `postcss.config.js` is not found `postcss-loader` will not be added and will not cause overhead.
 
+### Leverage long-term browser caching of style.css in production
```
Comment (timneutkens) [inline packages/next-css/readme.md]: This should be the main installation step if we change the filename to include the hash.

Comment (bhongy) [reply packages/next-css/readme.md]: Did you mean that we should incorporate it into the "Usage" section instead?

Comment (timneutkens) [reply packages/next-css/readme.md]: Yeah because otherwise people will create issues that the package doesn't work in production 😄

Comment (bhongy) [reply packages/next-css/readme.md]: Good point! Less confusion. Updated in 127db81. Please let me know what you think.

File: packages/next-css/readme.md [hunk 3]
```diff
@@ -137,28 +140,6 @@ Your exported HTML will then reflect locally scoped CSS class names.
 
 For a list of supported options, [refer to the webpack `css-loader` README](https://github.com/webpack-contrib/css-loader#options).
 
-```js
```
Comment (bhongy) [inline packages/next-css/readme.md]: Seems like this was a duplication? It's the same as the code example in the usage section and didn't seem like it is related to the content in line 138.

**Comment by mitchellhuang** [comment]
This PR works great @bhongy, I pulled out next-css into a separate repo for those that want to npm install it easily. https://github.com/mitchellhuang/next-css

**Comment by timneutkens** [comment]
Hashing is now the default behavior, see https://github.com/zeit/next-plugins/pull/228

**Pull request closed without merging**